weather/module.py

```diff
@@ -1,23 +1,157 @@
-from typing import Optional
+from typing import Optional, List
+import aiohttp
 
+import nextcord
 from nextcord.ext import commands
 
 import pie.database.config
-from pie import check, i18n, logger
+from pie import check, i18n, logger, utils
+from pie.i18n.database import GuildLanguage, MemberLanguage
 
 from .database import Place
 
-_ = i18n.Translator("modules/fun").translate
+translator = i18n.Translator("modules/fun")
+_ = translator.translate
 guild_log = logger.Guild.logger()
 config = pie.database.config.Config.get()
 
+# number of days to get forecast for (including current day, max is 3)
+NUM_OF_FORECAST_DAYS = 3
+# dict for getting the data from json easier (when you don't wan't some phase of day comment it)
+DAY_PHASES = {
+    "Morning": 2,
+    "Day": 4,
+    # "Evening": 6,
+    "Night": 7,
+}
+
 
 class Weather(commands.Cog):
     """Weather and forecast"""
 
-    def __init__(self, bot):
+    def __init__(self, bot: commands.Bot):
         self.bot = bot
 
+    def _translate_day_phase(self, ctx: commands.Context, day_phase: str) -> str:
+        if day_phase == "Morning":
+            return _(ctx, "Morning")
+        if day_phase == "Day":
+            return _(ctx, "Day")
+        if day_phase == "Evening":
+            return _(ctx, "Evening")
+        if day_phase == "Night":
+            return _(ctx, "Night")
+
+    def _get_useful_data(
+        self, all_data: dict, ctx: commands.Context, lang_preference: str
+    ) -> List[dict]:
+        """
+        example json: https://wttr.in/praha?lang=sk&format=j1
+        get useful data from json as list of individual days
+        """
+
+        # get individual days to extract data
+        weather = []
+        nearest_place = all_data["nearest_area"][0]["areaName"][0]["value"]
+        lang_preference = f"lang_{lang_preference}"
+        for i in range(NUM_OF_FORECAST_DAYS):
+            day = all_data["weather"][i]
+            day_dict = {
+                "date": day["date"],
+                "nearest_place": nearest_place,
+            }
+            day = day["hourly"]
+            for day_phase, hour in DAY_PHASES.items():
+                if lang_preference != "lang_en":
+                    lang_or_desc = lang_preference
+                else:
+                    lang_or_desc = "weatherDesc"
+                day_dict.update(
+                    {
+                        self._translate_day_phase(ctx, f"{day_phase}"): {
```

Comment:
Why is it `f"{day_phase}"` and not just `day_phase`?

Reply:
Oh sorry, forgot to rewrite it